.github/workflows/tests_linux_x86_nvidia_gpu.yml

```diff
@@ -0,0 +1,107 @@
+name: Tests::Linux::x86_64::Nvidia_GPU
+on:
+  release:
+  push:
+    branches:
+      - main
+  pull_request:
+
+env:
+  COVERAGE_FLAGS: "--cov=pennylane_lightning_kokkos --cov-report=term-missing --cov-report=xml:./coverage.xml --no-flaky-report -p no:warnings --tb=native" 
+  GCC_VERSION: 11
+  CI_CUDA_ARCH: 86
+
+concurrency:
+  group: gpu-test-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  cpptests:
+    runs-on:
+      - self-hosted
+      - ubuntu-22.04
+      - gpu
+
+    strategy:
+      max-parallel: 1
+
+    steps:
+      - uses: actions/setup-python@v4
+        name: Install Python
+        with:
+          python-version: '3.8'
+
+      - name: Remove Ubuntu unattended upgrades
+        run: sudo apt-get remove -y -q unattended-upgrades
+
+      - name: Install required packages
+        run: |
+          sudo apt-get update && sudo apt-get -y -q install gcc-${{ env.GCC_VERSION }} g++-${{ env.GCC_VERSION }} gcovr lcov
+          python -m pip install ninja cmake~=3.24.3
+      - name: Checkout PennyLane-Lightning-Kokkos
+        uses: actions/checkout@v3
+
+      - name: Validate GPU version and installed compiler
+        run: |
+          nvidia-smi
+          /usr/local/cuda-11.8/bin/nvcc --version
+      - name: Build and run unit tests
+        run: |
+            export PATH=/usr/local/cuda-11.8/bin:$PATH
+            export LD_LIBRARY_PATH=/usr/local/cuda-11.8/lib64:$LD_LIBRARY_PATH
+            cmake . -BBuild \
+              -DKokkos_ENABLE_CUDA=ON \
+              -DKokkos_ENABLE_SERIAL=ON \
+              -DPLKOKKOS_BUILD_TESTS=ON \
+              -G Ninja
+            cmake --build ./Build
+            ./Build/pennylane_lightning_kokkos/src/tests/runner_kokkos
+      - name: Cleanup
+        if: always()
+        run: |
+          rm -rf *
+          rm -rf .git
+          rm -rf .gitignore
+          rm -rf .github
+  pythontests:
+    runs-on:
+      - self-hosted
+      - ubuntu-22.04
+      - gpu
+
+    strategy:
+      max-parallel: 1
+
+    steps:
+      - uses: actions/setup-python@v4
+        name: Install Python
+        with:
+          python-version: '3.8'
+
+      - name: Install required packages
+        run: |
+          python -m pip install ninja cmake~=3.24.3 pytest pytest-mock flaky pytest-cov
```

Comment:
Why don't we `pip install -r requirements-ci.txt` here? Actually, `requirements-ci.txt` doesn't have the right packages at the moment so I'm wondering what it is used for. But should we not have the necessary stuff in some requirement file?

Reply:
Good idea! I think it is good for the moment. We can figure out a way to put these requirements together to make sure the `requirement_ci.txt `works for all the CI workflows later. How do you think?